tests/unit/s2n_quic_support_io_test.c

```diff
@@ -416,5 +429,96 @@ int main(int argc, char **argv)
+            EXPECT_SUCCESS(s2n_stuffer_write_uint8(&stuffer, TLS_KEY_UPDATE));
+            EXPECT_SUCCESS(s2n_stuffer_write_uint24(&stuffer, TEST_DATA_SIZE));
+            EXPECT_SUCCESS(s2n_stuffer_write_bytes(&stuffer, TEST_DATA, TEST_DATA_SIZE));
+
+            EXPECT_FAILURE_WITH_ERRNO(s2n_connection_process_post_handshake_message(conn), S2N_ERR_UNSUPPORTED_WITH_QUIC);
+        };
+
+        /* Test: successfully reads and processes post-handshake message */
+        {
+            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT),
+                    s2n_connection_ptr_free);
+            DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
+            EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(config, 1));
+            uint8_t session_ticket_cb_count = 0;
+            EXPECT_SUCCESS(s2n_config_set_session_ticket_cb(config, s2n_test_session_ticket_cb, &session_ticket_cb_count));
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+            conn->secure->cipher_suite = &s2n_tls13_aes_128_gcm_sha256;
+
+            DEFER_CLEANUP(struct s2n_stuffer stuffer = { 0 }, s2n_stuffer_free);
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&stuffer, 0));
+            EXPECT_SUCCESS(s2n_connection_set_io_stuffers(&stuffer, &stuffer, conn));
+
+            /* Construct ST handshake message */
+            EXPECT_SUCCESS(s2n_stuffer_write_bytes(&stuffer, ticket_message, sizeof(ticket_message)));
+
+            EXPECT_SUCCESS(s2n_connection_process_post_handshake_message(conn));
+
+            /* Callback was triggered */
+            EXPECT_EQUAL(session_ticket_cb_count, 1);
+        };
+
+        /* Test: successfully reads and processes fragmented post-handshake message */
+        for (size_t i = 1; i < sizeof(ticket_message); i++) {
+            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT), s2n_connection_ptr_free);
+            DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
+            EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(config, 1));
+            uint8_t session_ticket_cb_count = 0;
+            EXPECT_SUCCESS(s2n_config_set_session_ticket_cb(config, s2n_test_session_ticket_cb, &session_ticket_cb_count));
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+            conn->secure->cipher_suite = &s2n_tls13_aes_128_gcm_sha256;
+
+            DEFER_CLEANUP(struct s2n_stuffer stuffer = { 0 }, s2n_stuffer_free);
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&stuffer, 0));
+            EXPECT_SUCCESS(s2n_connection_set_io_stuffers(&stuffer, &stuffer, conn));
+
+            /* Mock receiving a fragmented handshake message */
+            EXPECT_SUCCESS(s2n_stuffer_write_bytes(&stuffer, ticket_message, i));
+
+            EXPECT_FAILURE_WITH_ERRNO(s2n_connection_process_post_handshake_message(conn), S2N_ERR_IO_BLOCKED);
+
```

Comment:
Note: I'm assuming you're deliberately leaving "blocked" out of this API because it's internal and not a one-way door? If it becomes possible for it to block on something other than S2N_BLOCKED_ON_READ you could add the field? Because theoretically a post-handshake message could someday block on a callback.